### ISSUE TITLE
Rollback doesn't remove tables created via `change()`

### ISSUE THREAD

**Issue by Stalinko** [cakephp/phinx#1307]
Just wasted half-hour because of this bug. The title is quite self-explanatory.

Make some migration which creates a table:

```
<?php

use Phinx\Migration\AbstractMigration;

class CreateTestTable extends AbstractMigration
{
    public function change()
    {
        $this->table('test')
            ->addColumn('name', 'string')
            ->save();
    }
}

```

Then do migrate and rollback. At rollback it will simply delete all columns but `id` and keep the table.

**Comment by chinpei215** [comment]
Probably, you would need to use `create()` instead of `save()`. Since `save()` does either `update()` or `create()`, I think it is not easy to rollback to the original state.

**Comment by Stalinko** [comment]
@chinpei215, you're absolutely right! Thanks.
Probably it would be nice to state that case in the docs, because the docs suggest to create tables via `save()` not `create()` (see https://book.cakephp.org/3.0/en/phinx/migrations.html#creating-a-table)

**Comment by chinpei215** [comment]
@Stalinko You are welcome. And probably you are right. I think the document should be fixed.

**Comment by lorenzo** [comment]
Closing as this will not be a requirement anymore when the internals refactoring branch is merged

**Comment by chinpei215** [comment]
I cannot confirm that this issue has been fixed. Though I think this is documentation issue, if you were planing to fix this issue by internal changes, please confirm this again.

**Comment by lorenzo** [comment]
@chinpei215 You mean that calling `create` or `update` is required instead of just calling `save`? I was quite certain that I fixed in 0.10

**Comment by chinpei215** [comment]
@lorenzo I mean that people need to call `create()` when creating tables. If you call `save()` when creating tables [like this example](https://book.cakephp.org/3.0/en/phinx/migrations.html#creating-a-table), those tables will be not removed after rolled back.

**Comment by lorenzo** [comment]
@chinpei215 is that still the case in 0.10?

**Comment by chinpei215** [comment]
@lorenzo Yes. If I ran the above example shown in the issue, the `test` table is not removed after rolled back:
```
mysql> desc test;
+-------+--------------+------+-----+---------+----------------+
| Field | Type         | Null | Key | Default | Extra          |
+-------+--------------+------+-----+---------+----------------+
| id    | int(11)      | NO   | PRI | NULL    | auto_increment |
| name  | varchar(255) | NO   |     | NULL    |                |
+-------+--------------+------+-----+---------+----------------+
2 rows in set (0.00 sec)

mysql> desc test;
+-------+---------+------+-----+---------+----------------+
| Field | Type    | Null | Key | Default | Extra          |
+-------+---------+------+-----+---------+----------------+
| id    | int(11) | NO   | PRI | NULL    | auto_increment |
+-------+---------+------+-----+---------+----------------+
1 row in set (0.00 sec)
```

**Comment by lorenzo** [comment]
crap, I need to restore those warnings, then. I though the need for this was gone :(

**Comment by chinpei215** [comment]
I think it is difficult to fix this issue. In my opinion, if we want to do that, probably we would need to change phinxlog schema, and mark when tables were created. I am not sure if it is a good idea. So I think it is a documentation issue.
As another option, we could throw warnings when tables were created by `save()`. (or throw errors? it breaks BC though).
